docs/chapters/traversal/index.md

```diff
@@ -0,0 +1,178 @@
+
+This allows Jina (and you) to query chunks and reference back to its root document at any `granularity` level.
+
+## Matches
+
+In a neural search system (and traditional retrieval system), matches are the expected documents returned from the system given the user query.
+In Jina, `Matches` could happen at the `root` level or any `chunk` level.
+
+To fully understand the concept of `Matches`, we introduce a new term, named `adjacency` (short for `a` in the diagram), which reflects the level of the document it is connected to.
+
+**NOTE: granularity and adjacency apply to both chunks and matches.**
+
+```python
+from jina import Document
+
+with Document() as root:
+    root.text = 'What is love? Oh baby do not hurt me.'
+print(root.adjacency)
+>>> 0
+# Initialise two documents and add as chunks to root.
+with Document() as chunk1:
+    chunk1.text = 'What is love?'
+    root.chunks.add(chunk1)
+with Document() as chunk2:
+    chunk1.text = 'Oh baby do not hurt me.'
+    root.chunks.add(chunk2)
+# Add a match document.
+with Document() as match:
+    # a match document semantically related to our root
+    match.text = 'What is love? Oh please do not hurt me.'
+    root.matches.add(match)
+print(len(root.matches))
+>>> 1
+print(root.matches[0].granularity)
+>>> 0
+print(root.matches[0].adjacency)
+>>> 1
+
+```
+
+![adjacency](img/adjacency.png)
+
+In the code snippet and diagram above, we initialized a `Document` as `root` with the text: *What is love? Oh, baby do not hurt me.*.
+And a `Document` with text *What is love? Oh please do not hurt me* was added as a match to the `root`.
+The matched document `match` is a document without any parents, so it stays at the same level as `root` with a granularity value of 0.
+Meanwhile, since `match` is the retrieved result from `root`, so the `adjacency` increased to 1.
+
+By default, the `root` node has an `adjacency` of 0, and the value increases by 1 when it hit a `match`.
+
+## Let's go deeper: Recursive Document Representation
+
+Till now, we've introduced `chunks` and `matches` **with a depth of 1**.
+While in a real-world scenario, things could be much more complicated than this.
+For instance, a `Chunk` could be further divided into small chunks, and a chunk at any level might have it's own `matches` at that level.
+
+![go-deeper](https://hanxiao.io/2020/08/28/What-s-New-in-Jina-v0-5/blog-post-v050-protobuf-documents.jpg)
+
+Jina has defined a recursive structure with **arbitrary width and depth** instead of a trivial bi-level structure.
+Roughly speaking, `chunks` can have the next level chunks and the same level matches; and so do matches.
+This could go on and on. The following figure illustrates this structure [Ref: New Features in Jina v0.5 You Should Know About](https://hanxiao.io/2020/08/28/What-s-New-in-Jina-v0-5/).
+
+![recursive](img/recursive.png)
+
+This recursive structure allows the Document to represent many real-world objects much comfortable than before.
```

Comment:
I think this sentence needs rewording, I would say something in the lines of:

This recursive structure provides Jina the flexibility to cover any complex usecase that may require search at different semantic units.

??

Reply:
I may also introduce the concept that some `Rankers` allow to "group", "bubble up" or "accumulate" scores from lower granularities to upper levels (which enables any combination of information)

Reply:
I agree with Joan, "comfortable" doesn't seem like the right word here.

Reply:
@JoanFM can you give an example of 
> some Rankers allow to "group", "bubble up" or "accumulate" scores from lower granularities to upper levels (which enables any combination of information)

Reply:
MinRanker or MaxRanker for instance; Chunk2DocRankers family